Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -171,9 +171,31 @@ open class BasicFormat: SyntaxRewriter {
       }
     }
 
-    return false
+    let syntaxToken: Syntax = Syntax(token)
+    switch syntaxToken.keyPathInParent {
+    case \StringLiteralExprSyntax.closeQuote:
+      return token.tokenKind == .multilineStringQuote
+    default:
+      return false
+    }
   }
 
+    /// Whether a trailing newline on `token` should be added.
+    open func requiresTrailingNewline(_ token: TokenSyntax) -> Bool {
```

Comment:
I think you will also need to check this in `previousTokenWillEndWithWhitespace`.